app/web/features/auth/notifications/PushNotificationPermission.tsx

```diff
@@ -0,0 +1,193 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import CustomColorSwitch from "components/CustomColorSwitch";
+import { Trans, useTranslation } from "i18n";
+import { AUTH } from "i18n/namespaces";
+import Sentry from "platform/sentry";
+import { useEffect, useState } from "react";
+import {
+  getVapidPublicKey,
+  registerPushNotificationSubscription,
+} from "service/notifications";
+import { theme } from "theme";
+import { arrayBufferToBase64 } from "utils/arrayBufferToBase64";
+import makeStyles from "utils/makeStyles";
+
+import { getCurrentSubscription } from "./notificationUtils";
+import PushNotificationDenied from "./PushNotificationDenied";
+
+const useStyles = makeStyles((theme) => ({
+  alert: {
+    marginBottom: theme.spacing(3),
+    marginTop: theme.spacing(2),
+  },
+  status: {
+    marginBottom: theme.spacing(2),
+  },
+  titleBox: {
+    display: "flex",
+    alignItems: "center",
+  },
+}));
+
+export default function PushNotificationPermission({
+  className,
+}: {
+  className: string;
+}) {
+  const { t } = useTranslation(AUTH);
+  const classes = useStyles();
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [isPushEnabled, setIsPushEnabled] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [shouldPromptAllow, setShouldPromptAllow] = useState<boolean>(false);
```

Comment:
Could you add a comment saying "whether to show the user instructions to click 'Allow' in their browser"